src/fault_condition.rs

```diff
@@ -31,42 +32,31 @@ impl FaultCondition {
              * the condition active flag and reset the last detection time.
              */
             self.monitoring_active = false;
-            self.condition_start_time = 0;
         } else {
-            let now = timer_ms.ms();
-
             if !self.monitoring_active {
                 /* We just detected a condition that may lead to a fault. Update
                  * the state to track that the condition is active and store the
                  * first time of detection.
                  */
                 self.monitoring_active = true;
-                self.condition_start_time = now;
+                self.timer.reset();
             }
 
-            // TODO - need to fix this ported logic
-            // panicked at 'attempt to subtract with overflow', src/fault_condition.rs:47:28
-            let duration = now - self.condition_start_time;
-
-            if duration >= max_duration {
+            if self.timer.wait().is_ok() {
```

Comment:
So, all of these checks now do a blocking wait for the time interval to pass?  Or does this `wait` call not actually blocking-wait but instead just read off a (possibly cached) value?

Reply:
They currently do not block in this context. The [Countdown trait](https://github.com/rust-embedded/embedded-hal/blob/99a0f4db7e05e799df5f1ef1c1628883b79addf0/src/timer.rs#L69) is often used by HAL crates in conjunction with the [`block!` macro](https://docs.rs/nb/0.1.1/nb/#blocking-mode). In this case we just poll the [timer bit](https://github.com/jonlamb-gh/stm32f767-hal/blob/devel/src/timer.rs#L79) on each iteration rather than wait for the event to occur.